plugins/lua/buildingplan/planneroverlay.lua

```diff
@@ -302,13 +317,17 @@ function ItemLine:init()
             frame={t=0, l=28},
             text={
                 {text=self:callback('get_filter_text'),
-                 pen=function() return gui.invert_color(COLOR_LIGHTCYAN, self.is_selected_fn()) end},
+                 width=item_filter_chars+2,
+                 rjustify=true,
+                 pen=function() return
+                         self:is_impossible() and COLOR_RED or
+                         gui.invert_color(COLOR_LIGHTCYAN, self.is_selected_fn()) end},
```

Comment:
I thought this would be 
```lua
gui.invert_color(self:is_impossible() and COLOR_RED or COLOR_LIGHTCYAN, self.is_selected_fn()) end
```
I'll have to check in game for which is appropriate

Reply:
I started with the one you suggested. But then only highlighted row show as red and the others show in the same orange as `[clear]`, which also does not change based on selection.